src/google-classroom-extractor/edfi_google_classroom_extractor/api/courses.py

```diff
@@ -20,6 +20,178 @@
+    List[str]
+        a list of alias names (scope omitted) filtered by the given scope
+    """
+    result: List[str] = []
+    try:
+        result = [
+            alias.split(":")[1] for alias in aliases if alias.split(":")[0] == scope
+        ]
+    except (ValueError, IndexError):
+        pass
+    return result
+
+
+def _find_first_with_edfilms_prefix(alias_names: List[str]) -> str:
+    """
+    Returns the first alias name in the list prefixed with the
+    Ed-Fi LMS indicator, or an empty string if none are found.
+
+    Parameters
+    ----------
+    alias_names: List[str]
+        a list of alias names
+
+    Returns
+    -------
+    str
+        The first alias name in the list prefixed with the
+        Ed-Fi LMS indicator, or an empty string if none are found.
+    """
+    return next(
+        (alias_name for alias_name in alias_names if alias_name.startswith("EdFiLMS")),
+        "",
+    )
+
+
+def _select_alias(aliases: List[str]) -> str:
+    """
+    Selects the most relevant course alias from a list of course aliases returned by the API
+
+    Parameters
+    ----------
+    aliases: List[str]
+        a list of course alias strings from the API, including the "d:" (domain scope)
+        or "p:" (project scope) prefix
+
+    Returns
+    -------
+    str
+        a (possibly empty) string representing the most relevant alias for the course.
+
+    Notes
+    -----
+    How to find the correct SISSectionIdentifier from CourseAlias results:
+
+    If there are no aliases:
+    SISSectionIdentifier = ""
+
+    Else if there is exactly one alias:
+    SISSectionIdentifier = alias
+
+    Else if there is a domain alias prefixed with "EdFiLMS":
+    SISSectionIdentifier = first domain alias prefixed with "EdFiLMS"
+
+    Else if there is a project alias prefixed with "EdFiLMS":
+    SISSectionIdentifier = first project alias prefixed with "EdFiLMS"
+
+    Else if there are domain aliases:
+    SISSectionIdentifier = first domain alias
+
+    Else if there are project aliases:
+    SISSectionIdentifier = first project alias
+
+    Else:
+    SISSectionIdentifier = ""
+    """
+    if len(aliases) == 0:
+        return ""
+
+    if len(aliases) == 1:
+        try:
+            _, alias_name = aliases[0].split(":")
+            return alias_name
+        except ValueError:
+            return ""
+
+    domain_alias_names: List[str] = _alias_names_for_scope(aliases, "d")
+    edfilms_domain_alias = _find_first_with_edfilms_prefix(domain_alias_names)
+    if edfilms_domain_alias != "":
+        return edfilms_domain_alias
+
+    project_alias_names: List[str] = _alias_names_for_scope(aliases, "p")
+    edfilms_project_alias = _find_first_with_edfilms_prefix(project_alias_names)
+    if edfilms_project_alias != "":
+        return edfilms_project_alias
+
+    if len(domain_alias_names) > 0:
+        return domain_alias_names[0]
+
+    if len(project_alias_names) > 0:
+        return project_alias_names[0]
+
+    return ""
```

Comment:
Another edge case without a test. Are we adequately testing what happens if no aliases at all have been setup?

Reply:
I have uploaded some aliases and the project aliases aren't coming through. For Spanish III I loaded "p:XYZ" and for English I I uploaded ["p:a1b2", "p:EdFiLMS.a1b2_"]. In both cases, the extractor output a blank value for the `SISSectionIdentifier`.

Reply:
Line 169 handles multiple unknown scopes without a known scope. Added test.

Reply:
The GC API isn't returning any aliases for Spanish III. Maybe because the course is status ARCHIVED?

Reply:
I suspect a permission problem, because I'm able to access that from the [API tester](https://developers.google.com/classroom/reference/rest/v1/courses.aliases/list)
![image](https://user-images.githubusercontent.com/9324390/126703135-6ec40b47-12fa-41d6-bd3d-b35c38b83d39.png)

Reply:
Removed project scope support, doesn't look relevant.